tests/unit/test_tempest_extremes.py

```diff
@@ -502,3 +601,243 @@ def test_stitch_nodes_failure(self, mocker) -> None:
             RuntimeError, match="StitchNodes failed with a non-zero exit code"
         ):
             tracker.stitch_nodes()
+
+    @pytest.fixture
+    def mock_gfdl_file(self, tmp_path):
+        """Fixture to create a mock GFDL file with two tracks."""
+        file_path = tmp_path / "tracks_out_gfdl.txt"
+        file_path.write_text(
+            "start\t45\t1950\t1\t1\t3\n"
+            "\t164\t332\t57.832031\t-12.070312\t1.005377e+05\t0.000000e+00\t1950\t1\t1\t3\n"
+            "\t163\t332\t57.480469\t-12.070312\t1.005820e+05\t0.000000e+00\t1950\t1\t1\t6\n"
+            "start\t12\t1950\t1\t2\t0\n"
+            "\t843\t275\t296.542969\t-25.429688\t9.970388e+04\t2.633214e+02\t1950\t1\t2\t0\n"
+            "\t850\t266\t299.003906\t-27.539062\t9.989988e+04\t6.951086e+01\t1950\t1\t2\t6\n"
+        )
+        return str(file_path)
+
+    @pytest.fixture
+    def mock_csv_file(self, tmp_path):
+        """Fixture to create a mock CSV file with two tracks."""
+        file_path = tmp_path / "tracks_out_csv.txt"
+        file_path.write_text(
+            "track_id,year,month,day,hour,i,j,lon,lat,psl,orog\n"
+            "0,1950,1,1,3,164,332,57.832031,-12.070312,1.005377e+05,0.000000e+00\n"
+            "0,1950,1,1,6,163,332,57.480469,-12.070312,1.005820e+05,0.000000e+00\n"
+            "1,1950,1,2,0,843,275,296.542969,-25.429688,9.970388e+04,2.633214e+02\n"
+            "1,1950,1,2,6,850,266,299.003906,-27.539062,9.989988e+04,6.951086e+01\n"
+        )
+        return str(file_path)
+
+    @pytest.fixture
+    def mock_csvnohead_file(self, tmp_path):
+        """Fixture to create a mock CSV file without a header and with two tracks."""
+        file_path = tmp_path / "tracks_out_csvnohead.txt"
+        file_path.write_text(
+            "0,1950,1,1,3,164,332,57.832031,-12.070312,1.005377e+05,0.000000e+00\n"
+            "0,1950,1,1,6,163,332,57.480469,-12.070312,1.005820e+05,0.000000e+00\n"
+            "1,1950,1,2,0,843,275,296.542969,-25.429688,9.970388e+04,2.633214e+02\n"
+            "1,1950,1,2,6,850,266,299.003906,-27.539062,9.989988e+04,6.951086e+01\n"
+        )
+        return str(file_path)
+
```

Comment:
```suggestion

    def expected_tracks_data(i_track: int, var_names: list[str]):
        grid_i = [[164, 163], [843, 850]]
        grid_j = [[332, 332], [275, 266]]
        var_1 = [[57.832031, 57.480469], [296.542969, 299.003906]]
        var_2 = [[-12.070312, -12.070312], [-25.429688, -27.539062]]
        var_3 = [[1.005377e05, 1.005820e05], [9.970388e04, 9.989988e04]]
        var_4 = [[0.0, 0.0], [2.633214e02, 6.951086e01]]
        return [{
            "grid_i": grid_i[i_track][i_point],
            "grid_j": grid_j[i_track][i_point],
            "variables": {
                var_names[0]: var_1[i_track][i_point],
                var_names[1]: var_2[i_track][i_point],
                var_names[2]: var_3[i_track][i_point],
                var_names[3]: var_4[i_track][i_point],
            }
        } for i_point in [0, 1]]

```
I would propose using the above function (or something like it) to simplify the expected_tracks parameters and make the test a bit more comprehensible.